### PR TITLE
Added timestamp compare

### PR DESCRIPTION
**Pull request by ronykoz**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24845

## Description
Added timestamp compare

image

### CODE REVIEW

File: Packs/CommonScripts/Scripts/TimeStampCompare/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## [Unreleased]
+
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/TimeStampCompare/CHANGELOG.md:2]: how did it pass validate?

File: Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+DT_STRING = "TimeStampCompare(val.TestedTime && val.TestedTime == obj.TestedTime && " \
+            "val.ComparedTime && val.ComparedTime == obj.ComparedTime)"
+
+
+def time_stamp_compare_command(args):
+    tested_time = args.get('tested_time')
+    values_to_compare = argToList(args.get('values_to_compare'))
+
+    results = []
+    parsed_tested_time = parse(tested_time)
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.py:22]: what about cases like 2/11/2020 > 1/12/2020? shouldn't we add date-format args?

File: Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    for compared_time in values_to_compare:
+        parsed_compared_time = parse(compared_time)
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.py:24]: same here

File: Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.py (hunk 2 above)
Comment (David-BMS) [reply Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.py:22]: @dantavori exactly we need to cover all possible timestamp formats

File: Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+name: TimeStampCompare
+outputs:
+- contextPath: TimeStampCompare.TestedTime
+  description: The tested time
+  type: Date
+- contextPath: TimeStampCompare.ComapredTime
+  description: The compared time
+  type: Date
+- contextPath: TimeStampCompare.Result
+  description: Whether the tested time was before/after/equal as the comapred time.
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.yml:28]: The tested time is something the user can configure such as one month ago (lets say by using the !GetTime command)
The values_to_compare are are the values that we want to check if they are before/after/equal the tested_time

See the context example I provided in the issue
Example Context
FilterByTime.0.TestedTime 01-01-2020 00:00:00
FilterByTime.0.ComparedTime 02-01-2020
FilterByTime.0.Value after

[the remark this answers is not in the document]
Comment (ronykoz) [reply Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.yml:28]: @David-BMS why is it after?

File: Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+- default: false
+  description: Values to compare the against the tested_time, we are checking each value and determining whether it's before/after/equal to the tested_time.
+  isArray: true
+  name: values_to_compare
+  required: true
+  secret: false
+- default: false
+  description: Time format of the times you entered, by default we are on automatic parsing - this should be used for cases like DD/MM/YYYY
+  isArray: false
+  name: time_format
```
Comment (David-BMS) [inline Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.yml:17]: Does this apply to both the tested_time and values_to_compare?

File: Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  required: true
+  secret: false
+- default: false
+  description: Values to compare the against the tested_time, we are checking each value and determining whether it's before/after/equal to the tested_time.
+  isArray: true
+  name: values_to_compare
+  required: true
+  secret: false
+- default: false
+  description: Time format of the times you entered, by default we are on automatic parsing - this should be used for cases like DD/MM/YYYY
```
Comment (David-BMS) [inline Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.yml:15]: Best to add a few examples of the acceptable formats

**Comment by ronykoz** [comment]
@kirbles19 could you please go over this one file:
https://github.com/demisto/content/blob/added-timestamp-compare/Packs/CommonScripts/Scripts/TimeStampCompare/TimeStampCompare.yml

**Comment by kirbles19** [comment]
@ronykoz Done.

**Pull request merged**